lib/metasploit/framework/data_service/remote/http/remote_workspace_data_service.rb

```diff
@@ -3,55 +3,49 @@
 module RemoteWorkspaceDataService
   include ResponseDataHelper
 
-  # TODO: should counts be a flag in query data for the workspaces resource?
-  WORKSPACE_COUNTS_API_PATH = '/api/v1/workspaces/counts'
   WORKSPACE_API_PATH = '/api/v1/workspaces'
   WORKSPACE_MDM_CLASS = 'Mdm::Workspace'
-  DEFAULT_WORKSPACE_NAME = 'default'
-
-  def find_workspace(workspace_name)
-    workspace = workspace_cache[workspace_name]
-    return workspace unless (workspace.nil?)
-
-    workspace = json_to_mdm_object(self.get_data(WORKSPACE_API_PATH, {:workspace_name => workspace_name}), WORKSPACE_MDM_CLASS).first
-    workspace_cache[workspace_name] = workspace
-  end
 
   def add_workspace(workspace_name)
     response = self.post_data(WORKSPACE_API_PATH, {:workspace_name => workspace_name})
-    json_to_mdm_object(response, WORKSPACE_MDM_CLASS, nil)
+    json_to_mdm_object(response, WORKSPACE_MDM_CLASS, nil).first
   end
 
   def default_workspace
-    find_workspace(DEFAULT_WORKSPACE_NAME)
+    json_to_mdm_object(self.get_data(WORKSPACE_API_PATH, nil, {:name => Msf::DBManager::Workspace::DEFAULT_WORKSPACE_NAME}), WORKSPACE_MDM_CLASS, [])
```

Comment:
Style guide: [Use the Ruby 1.9 hash literal syntax when your hash keys are symbols.](https://github.com/bbatsov/ruby-style-guide#hash-literals). Also see hash style above in `add_workspace` method.